Source/DafnyCore/AST/Expressions/Expressions.cs

```diff
@@ -1928,21 +1929,25 @@ public enum AccumulationOperand { None, Left, Right }
     Contract.Invariant(E1 != null);
   }
 
-  public BinaryExpr(IToken tok, Opcode op, Expression e0, Expression e1)
+  public BinaryExpr(IToken tok, Opcode op, Expression e0, Expression e1, IToken prefixOp)
```

Comment:
```suggestion
  public BinaryExpr(IToken tok, Opcode op, Expression e0, Expression e1, IToken prefixOp = null)
```
Wouldn't this be less invasive?

Reply:
It would be, and that's how I originally wrote it, but @MikaelMayer pointed out that it would make it easier to make mistakes during cloning by not cloning the `prefixOp`. Does avoiding those mistakes outweigh the extra change complexity? I'm undecided.